packages/ui-react/src/containers/AccountModal/forms/PersonalDetails.tsx

```diff
@@ -45,16 +46,20 @@ const PersonalDetails = () => {
   }, [navigate, location, accessModel, hasMediaOffers]);
 
   useEffect(() => {
-    if (data && (!data.isCaptureEnabled || !data.shouldCaptureBeDisplayed)) nextStep();
+    if (!data) return;
 
-    if (data && questions) {
+    if (!data.isCaptureEnabled || !data.shouldCaptureBeDisplayed) {
+      nextStep();
+    }
+
+    if (questions) {
       setQuestionValues(Object.fromEntries(questions.map((question) => [question.key, ''])));
     }
   }, [data, nextStep, questions]);
 
   const initialValues: PersonalDetailsFormData = {
-    firstName: '',
-    lastName: '',
+    firstName: (fields.firstNameLastName?.answer as Record<string, string | null>)?.firstName ?? '',
+    lastName: (fields.firstNameLastName?.answer as Record<string, string | null>)?.lastName ?? '',
```

Comment:
Very nice changes @mirovladimitrovski! This looks clean

I'm curious (not for now) if we can move all `useForm` hooks to the component like this. This ensures that the `initialValues` are defined and makes a lot of props redundant. Initially, I thought the form logic belonged to the container level, but this seems fine. 

The only feedback remaining is the casting in the above code. It might be minor because we should be able to assume that `fields.firstNameLastName.answer` is an object with multiple values, I prefer to ensure that's the case.

In a refactor PR we did that has not landed yet, I only use `CustomFormField` types, which makes this a bit simpler, but perhaps we can add a simple util that resolves/extracts the value from a union type.

```ts
  const getCaptureValue = (field?: CleengCaptureField | CleengCaptureQuestionField, answerKey?: string) => {
    if (typeof field?.answer === 'string') return field.answer;
    if (typeof field?.answer === 'object' && field.answer !== null && !!answerKey) return field.answer[answerKey] ?? '';
    return '';
  };
```

```suggestion
    firstName: getCaptureValue(fields.firstNameLastName, 'firstName'),
    lastName: getCaptureValue(fields.firstNameLastName, 'lastName'),
```

Reply:
Yeah, I don't like the casting either, especially that it's the same casting written twice. I reduced it to one statement in my latest commit. I think we're good to go for now.